### ISSUE TITLE
Create an article "Immutable message and serialization"

### ISSUE THREAD

**Issue by alexvaluyskiy** [akkadotnet/akka.net#2675]
Base on these articles
http://getakka.net/docs/concepts/messages
http://akkanetdocfx.azurewebsites.net/articles/actors/untyped-actor-api.html#messages-and-immutability

**Comment by Danthar** [comment]
You mean as a document under "Concepts" ? Or something else ?

**Comment by alexvaluyskiy** [comment]
Yes, in "concepts". I think we should have a recommendation for messages. For example
- **DO** create immutable messages
- **AVOID** use complex types which should not be designed for serialization, like `Lazy`, `StringBuilder`
- **AVOID** use exceptions in your messages, unless your serializer supports them
- **AVOID** use `object` type in your messages.
- **AVOID** circular references in your messages
- **DO NOT** create more than one constructor in a message. You serializer-provided attributes to mark a default constructor
